src/array.c

```diff
@@ -695,16 +693,9 @@ static void NOINLINE array_try_unshare(jl_array_t *a)
     }
 }
 
-static size_t limit_overallocation(jl_array_t *a, size_t alen, size_t newlen, size_t inc)
+static size_t array_grow_newsize(size_t curlen)
 {
-    // Limit overallocation to jl_arr_xtralloc_limit
-    size_t es = a->elsize;
-    size_t xtra_elems_mem = (newlen - a->offset - alen - inc) * es;
-    if (xtra_elems_mem > jl_arr_xtralloc_limit) {
-        // prune down
-        return alen + inc + a->offset + (jl_arr_xtralloc_limit / es);
-    }
-    return newlen;
+    return curlen <= 10 ? curlen * 2 : curlen * 1.5;
```

Comment:
Not that it matters much, but in #16305 I used `((curlen*3)>>1)` to avoid floating-point conversion here.